### PR TITLE
Fixes and cleanup

### PR DESCRIPTION
**Pull request by hill-a**
- Fixed ```tf.session().__enter__()``` being used, rather than ```sess = tf.session()``` and passing the session to the objects
- Fixed uneven scoping of TensorFlow Sessions throughout the code
- Fixed rolling vecwrapper to handle observations that are not only grayscale images
- Fixed deepq saving the environment when trying to save itself
- Fixed ```ValueError: Cannot take the length of Shape with unknown rank.``` in ```acktr```, when running ```run_atari.py``` script.
- Fixed calling baselines sequentially no longer creates graph conflicts
- Fixed mean on empty array warning with deepq
- Fixed kfac eigen decomposition not cast to float64, when the parameter use_float64 is set to True
- Fixed Dataset data loader, not correctly resetting id position if shuffling is disabled
- Fixed ```EOFError``` when reading from connection in the ```worker``` in ```subproc_vec_env.py```
- fixed ```behavior_clone``` weight loading and saving for GAIL
- Avoid taking root square of negative number in `trpo_mpi.py`
- Removed some duplicated code (a2cpolicy, trpo_mpi)
- Removed unused, undocumented and crashing function ```reset_task``` in ```subproc_vec_env.py```
- Reformated code to PEP8 style
- Documented all the codebase
- Added atari tests
- Added logger tests

Missing: tests for acktr continuous (+ HER, gail but they rely on mujoco...)
TODO:
- [x] rename variables to be inline with PEP8
- [x] document the code correctly, and in the same style
- [x] add more tests

### CODE REVIEW

File: baselines/acktr/kfac.py [hunk 1]
```diff
@@ -1,16 +1,45 @@
...
+from baselines.acktr.kfac_utils import detect_min_val, factor_reshape, gmatmul
+
 KFAC_OPS = ['MatMul', 'Conv2D', 'BiasAdd']
 KFAC_DEBUG = False
 
 
-class KfacOptimizer():
-
-    def __init__(self, learning_rate=0.01, momentum=0.9, clip_kl=0.01, kfac_update=2, stats_accum_iter=60, full_stats_init=False, cold_iter=100, cold_lr=None, async=False, async_stats=False, epsilon=1e-2, stats_decay=0.95, blockdiag_bias=False, channel_fac=False, factored_damping=False, approxT2=False, use_float64=False, weight_decay_dict={},max_grad_norm=0.5):
+# TODO: fix missing/inacurate comments
```
Comment (araffin) [inline baselines/acktr/kfac.py]: @kalifou What is missing ? blockdiag_bias ?

File: baselines/acktr/run_atari.py [hunk 2]
```diff
@@ -1,23 +1,34 @@
...
+    """
+    train an ACKTR model on atari
+
+    :param env_id: (str) Environment ID
+    :param num_timesteps: (int) The total number of samples
+    :param seed: (int) The initial seed for training
+    :param num_cpu: (int) The number of cpu to train on
+    """
     env = VecFrameStack(make_atari_env(env_id, num_cpu, seed), 4)
-    policy_fn = partial(CnnPolicy, one_dim_bias=True)
```
Comment (araffin) [inline baselines/acktr/run_atari.py]: @hill-a  one_dim_bias does not exist, no ?

File: baselines/acktr/kfac.py (hunk 1 above)
Comment (hill-a) [reply baselines/acktr/kfac.py]: This was from me, I was not sure if the documentation is accurate or not

File: baselines/acktr/run_atari.py (hunk 2 above)
Comment (hill-a) [reply baselines/acktr/run_atari.py]: it is used for the conv layers of the CNN policies

https://github.com/hill-a/stable-baselines/blob/fixes_cleanup/baselines/a2c/utils.py#L95

Comment (araffin) [reply baselines/acktr/run_atari.py]: So you removed it because it was buggy ?

File: setup.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
           'Python {}. The installation will likely fail.'.format(sys.version_info.major))
 
 
-setup(name='baselines',
+setup(name='stable_baselines',
```
Comment (araffin) [inline setup.py]: To be changed before PR to OpenAI

File: baselines/acktr/run_atari.py (hunk 2 above)
Comment (hill-a) [reply baselines/acktr/run_atari.py]: good point, although I think it is broken on this branch, I can add it to the refactoring branch

File: baselines/acktr/acktr_cont.py [hunk 4]
```diff
@@ -1,93 +1,119 @@
...
+            "reward": np.array(rewards), "action": np.array(actions),
+            "action_dist": np.array(action_dists), "logp": np.array(logps)}
 
-def learn(env, policy, vf, gamma, lam, timesteps_per_batch, num_timesteps,
-    animate=False, callback=None, desired_kl=0.002):
 
+def learn(env, policy, value_fn, gamma, lam, timesteps_per_batch, num_timesteps,
+          animate=False, callback=None, desired_kl=0.002):
+    """
+    Learns a Kfac model
```
Comment (hill-a) [inline baselines/acktr/acktr_cont.py]: ACKTR model

File: baselines/gail/trpo_mpi.py [hunk 5]
```diff
@@ -145,83 +210,110 @@ def learn(env, policy_func, reward_giver, expert_dataset, rank,
...
-        MPI.COMM_WORLD.Allreduce(x, out, op=MPI.SUM)
+    def allmean(arr):
+        assert isinstance(arr, np.ndarray)
+        out = np.empty_like(arr)
+        MPI.COMM_WORLD.Allreduce(arr, out, op=MPI.SUM)
         out /= nworkers
         return out
 
-    U.initialize()
+    if using_gail:
```
Comment (hill-a) [inline baselines/gail/trpo_mpi.py]: remove useless gail flags

**Pull request merged**